### PR TITLE
Fixes issue #6303 Intellisense Soft-Selection is confusing

### PR DESCRIPTION
**Pull request by mkrueger**

### CODE REVIEW

**Comment by slluis** [comment]
@iainx can you please verify the fix?

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/DelegateCompletionProvider.cs [hunk 1]
```diff
@@ -297,7 +297,7 @@ void AddDelegateHandlers (CompletionContext context, SyntaxNode parent, Semantic
 			// item.CompletionCategory = category;
 			if (!context.Items.Any (i => i.DisplayText == item.DisplayText)) {
 				context.AddItem (item);
-				context.SuggestionModeItem = item;
+				item.WithRules (item.Rules.WithMatchPriority (int.MaxValue));
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/DelegateCompletionProvider.cs]: This does nothing?

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/DelegateCompletionProvider.cs [hunk 2]
```diff
@@ -273,7 +273,7 @@ void AddDelegateHandlers (CompletionContext context, SyntaxNode parent, Semantic
 					//item.CompletionCategory = category;
 					if (!context.Items.Any (i => i.DisplayText == item.DisplayText)) {
 						context.AddItem (item);
-						context.SuggestionModeItem = item;
+						item.WithRules (item.Rules.WithMatchPriority (int.MaxValue));
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/DelegateCompletionProvider.cs]: This does nothing?

**Comment by slluis** [comment]
@Therzok can you verify?

**Comment by Therzok** [comment]
Still seems to soft select for me.

**Comment by Therzok** [comment]
softsel

**Comment by mkrueger** [comment]
That soft selection is by design you need to type at least 1 char and that's fixed.

**Comment by Therzok** [comment]
Yeah, you're right.

**Pull request merged**